internal/fs/fs.go

```diff
@@ -2058,6 +2061,60 @@ func (fs *fileSystem) checkDirNotEmpty(dir inode.BucketOwnedDirInode, name strin
 	return nil
 }
 
+// Rename an old folder to a new folder in hierarchical bucket. If the new folder already
+// exists and is non-empty, return ENOTEMPTY.
```

Comment:
What specific error (if any) do we expect if the old directory does not exist, or has a file open?

Reply:
We are returning manual errors and not fuse errors directly on those cases. renameDir() function also mentioned the same info.